lib/arboreal/class_methods.rb

```diff
@@ -22,6 +23,28 @@ def populate_root_materialized_paths
       connection.update("UPDATE #{table_name} SET materialized_path = '-' WHERE parent_id IS NULL")
     end
 
+    def populate_root_relation
+      if connection.adapter_name =~ /mysql/i
+        connection.update(update_root_relation_sql)
+      else
+        roots.update_all(root_ancestor_id: nil)
+        roots.find_each { |root| root.descendants.update_all(root_ancestor_id: root.id) }
+      end
+    end
+
+    def update_root_relation_sql
+      sql = <<-SQL
+        UPDATE _arboreals_
+        SET root_ancestor_id =
+          IF(
+            materialized_path = '-',
+            NULL,
+            SUBSTRING_INDEX(SUBSTRING_INDEX(materialized_path, '-', 2), '-', -1)
+          )
+      SQL
+      sql.gsub("_arboreals_", table_name).squish
```

Comment:
Could also use `quote_table_name` http://apidock.com/rails/v3.2.3/ActiveRecord/ConnectionAdapters/PostgreSQLAdapter/quote_table_name to be paranoid.